descriptors/ModuleDescriptor-template.json

```diff
@@ -1056,6 +1056,33 @@
         }
       ]
     },
+    {
+      "id": "bound-with-parts",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["GET"],
+          "pathPattern": "/bound-with-parts",
+          "permissionsRequired": ["inventory-storage.bound-with-parts.collection.get"]
+        }, {
+          "methods": ["GET"],
+          "pathPattern": "/bound-with-parts/{id}",
+          "permissionsRequired": ["inventory-storage.bound-with-parts.item.get"]
+        }, {
+          "methods": ["POST"],
+          "pathPattern": "/bound-with-parts",
+          "permissionsRequired": ["inventory-storage.bound-with-parts.item.post"]
+        }, {
+          "methods": ["PUT"],
+          "pathPattern": "/bound-with-parts/{id}",
+          "permissionsRequired": ["inventory-storage.bound-with-parts.item.put"]
+        }, {
+          "methods": ["DELETE"],
+          "pathPattern": "/bound-with-parts/{id}",
+          "permissionsRequired": ["inventory-storage.bound-with-parts.item.delete"]
+        }
+      ]
+    },
```

Comment:
As I understand it, this interface is specific to an aspect of inventory. I think it could be useful if the paths used in the interface (and maybe the id) reflected that e.g. `inventory-storage/bound-with-parts` if this is intended to specifically be a storage level interface.

Are there intended to be endpoints for this information in business logic interfaces or is the intention that is all hidden behind another of the inventory business logic interfaces?

Reply:
The expected usages of /inventory-storage/bound-with-parts in business logic interfaces are:

* existing API /inventory/instances  will query inventory-storage/bound-with-parts to set the intance.isBoundWith flag property (so hidden behind)
* existing API /inventory/items will query inventory-storage/bound-with-parts to set the item.isBoundWith flag property (hidden behind)
* new API /inventory/bound-with-items  will GET  bound-with item(s) for a holdingsRecordId.  It would respond with objects on the regular Inventory Item schema. I would be used by ui-inventory to populate the item list under the holdings record in the Instance detail view, since said items would not be found by the standard way of querying the /inventory/items API by holdingsRecordId   (so that's a BL endpoint of sorts but not a straight forward propagation of the storage end-point)
* I'm imagining at least one other API  /inventory/bound-withs   which would return composite bound-with objects to be used by the UI for displaying the entire bound-with or multiple elements hereof. The composite would thus hold: The item, all the holdings, and all the Instances of the bound-with. There are UI mock-ups showing lists of titles in this fashion and compiling that information would not be trivial for the UI. 
* I haven't quite thought through what APIs might be useful for editing bound-withs through the UI. The raw /inventory-storage/bound-with-parts might even suffice, I don't yet know.

Reply:
> existing API /inventory/instances will query inventory-storage/bound-with-parts to set the intance.isBoundWith flag property (so hidden behind)

Why is it important that an instance be bound-with something else? Surely some copies of an instance might be bound-with copies of another instance, whilst other might not be.

> new API /inventory/bound-with-items will GET bound-with item(s) for a holdingsRecordId. It would respond with objects on the regular Inventory Item schema. I would be used by ui-inventory to populate the item list under the holdings record in the Instance detail view, since said items would not be found by the standard way of querying the /inventory/items API by holdingsRecordId (so that's a BL endpoint of sorts but not a straight forward propagation of the storage end-point)

Is this instead of introducing an API for holdings and including that information in the response to that API?

Reply:
Yes (some copies of an instance might be bound-with while others might not). This is a question for the Inventory PO and the GBV based working group) 

No, I have not thought of it as an alternative to an Inventory holdings API. I suppose. I reckon you mean, that if there was a holdings API in Inventory, it could conceivably retrieve both bound-with items and non-bound-with items behind the seems and serve them up in the composite?  That would be possible. It would require slightly bigger changes in the Inventory UI I suppose.

Reply:
Hmm, I changed the API path to /inventory-storage/bound-with-parts last week, but not sure if that becomes clear. The code snippets in the conversation doesn't seem to reflect the changes made - except for the flag "Outdated".  But code was amended anyways.

Reply:
> Yes (some copies of an instance might be bound-with while others might not). This is a question for the Inventory PO and the GBV based working group I think. They asked for flags anyway - alerts of a sort.

Do you need an answer to that question prior to designing the inventory API changes? I would think so, given that it might mean that they don't want this indicated on the instance (which describes an abstract notion that isn't actually bound to anything)

> No, I have not thought of it as an alternative to an Inventory holdings API. I reckon you mean that if there was a holdings API in Inventory, it could conceivably retrieve both bound-with items and non-bound-with items behind the seems and serve them up in a composite of holdings and items? That would be possible. It would possibly require slightly bigger changes in the Inventory UI.

Yes, that is what I was thinking. It was based upon the assumption that it was holdings where information about bound-with would be presented, given that it is these kinds of records we chose to represent that.